kuma/javascript/src/components/section-link.jsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { render } from 'react-dom';
+
+import { gettext } from '../l10n.js';
+
+import Anchor from '../icons/anchor.svg';
+
+/**
+ * Returns a section anchor element with a nested SVG icon
+ * @param {Object} heading - The current heading DOM node
+ * @returns section anchor element as an HTML element
+ */
+export default function sectionAnchor(heading) {
+    const anchor = document.createElement('a');
+    anchor.href = `#${heading.id}`;
+    anchor.classList.add('section-link');
+    // heading.innerText is the current heading and will already be translated
+    anchor.setAttribute('aria-label', gettext('Link to ') + heading.innerText);
+    // render the SVG into the anchor element
+    render(<Anchor />, anchor);
```

Comment:
Any idea if this is expensive?

Reply:
I do not and was wondering about that. But this should happen on the server no? I did not notice anything that suggested this is not a good idea, but I honestly don't know.

Reply:
Effects aren't run in SSR React. Also, the reason it's affect is because you want to to trigger when you change the document. I.e. clicking on an internal link and doing a pushState move.

Reply:
Ah, I see. Not sure how we can test whether it will be/is expensive. Another options are https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup, but seemed like too much for what this is doing. I could definitely be wrong though